tests/ablateLibrary/mathFunctions/parsedNestedTests.cpp

```diff
@@ -0,0 +1,146 @@
+#include <map>
+#include <memory>
+#include "gtest/gtest.h"
+#include "mathFunctions/functionFactory.hpp"
+#include "mathFunctions/parsedNested.hpp"
+#include "parser/mockFactory.hpp"
+#include "parser/registrar.hpp"
+
+namespace ablateTesting::mathFunctions {
+
+TEST(ParsedNestedTests, ShouldBeCreatedFromRegistar) {
+    // arrange
+    std::shared_ptr<ablateTesting::parser::MockFactory> mockFactory = std::make_shared<ablateTesting::parser::MockFactory>();
+    const std::string expectedClassType = "ablate::mathFunctions::ParsedNested";
+    EXPECT_CALL(*mockFactory, GetClassType()).Times(::testing::Exactly(1)).WillOnce(::testing::ReturnRef(expectedClassType));
```

Comment:
What does EXPECT_CALL do?

Reply:
It is part of the gMock library.  It tells the mockFactory to expect a certain call with specific arguments (GetClassType()).  It sets how many times it should be called and what it will return.  The test fails if these requirements are not done.  It makes it easier to test the parsedNested class without relying on a specific factory implementation.